frontend/src/v5/ui/components/shared/drawingRevisionDetails/drawingRevisionDetails.component.tsx

```diff
@@ -0,0 +1,133 @@
+import { uploadToDrawing } from '@/v5/ui/routes/dashboard/projects/drawings/uploadDrawingRevisionForm/uploadDrawingRevisionForm.helpers';
+import { SkeletonListItem } from '../revisionDetails/components/skeletonListItem/skeletonListItem.component';
+import { RevisionsListItem } from '../revisionDetails/components/revisionsListItem/revisionsListItem.component';
+import { RevisionsListHeaderLabel } from '../revisionDetails/components/revisionsListHeaderLabel/revisionsListHeaderLabel.component';
+import {
+	Container,
+	RevisionsListHeaderContainer,
+	RevisionsListItemWrapper,
+	RevisionsList,
+	RevisionsListEmptyWrapper,
+	RevisionsListEmptyContainer,
+	RevisionsListEmptyText,
+} from '../revisionDetails/revisionDetails.styles';
+import { getRevisionFileUrl } from '@/v5/services/api/drawingRevisions';
+import { selectHasCollaboratorAccess } from '@/v5/store/drawings/drawings.selectors';
+import { getState } from '@/v4/modules/store';
+
+interface IDrawingRevisionDetails {
+	drawingId: string;
+	revisionsCount: number;
+	status?: DrawingUploadStatus;
+}
+export const DrawingRevisionDetails = ({ drawingId, revisionsCount, status }: IDrawingRevisionDetails): JSX.Element => {
+	const teamspace = TeamspacesHooksSelectors.selectCurrentTeamspace();
+	const project = ProjectsHooksSelectors.selectCurrentProject();
+	const isLoading = DrawingRevisionsHooksSelectors.selectIsPending(drawingId);
+	const revisions = DrawingRevisionsHooksSelectors.selectRevisions(drawingId)
+		.sort((a, b) => b.timestamp.getTime() - a.timestamp.getTime());
+	const selected = revisions.findIndex((r) => !r.void);
+
+	const handleDownloadRevision = (revisionId) => {
+		window.location.href = getRevisionFileUrl(teamspace, project, drawingId, revisionId);
+	};
+
+	useEffect(() => {
+		if (!revisions.length) {
+			DrawingRevisionsActionDispatchers.fetch(teamspace, project, drawingId);
+		}
+	}, []);
+
+	if (!revisionsCount && !isLoading && !revisions.length) {
+		return (
+			<RevisionsListEmptyWrapper>
+				<RevisionsListEmptyContainer>
+					{
+						canUploadToBackend(status) ? (
+							<>
+								<RevisionsListEmptyText>
+									<FormattedMessage id="drawings.revisions.emptyMessage" defaultMessage="You haven’t added any Files." />
+								</RevisionsListEmptyText>
+								<Button
+									startIcon={<ArrowUpCircleIcon />}
+									variant="contained"
+									color="primary"
+									onClick={() => uploadToDrawing(drawingId)}
+								>
+									<FormattedMessage id="drawings.revisions.uploadFile" defaultMessage="Upload File" />
+								</Button>
+							</>
+						) : (
+							<RevisionsListEmptyText>
+								<FormattedMessage id="drawings.revisions.emptyMessageBusy" defaultMessage="Your files are being processed at this moment, please wait before creating new revisions for this drawing." />
+							</RevisionsListEmptyText>
+						)
+					}
+				</RevisionsListEmptyContainer>
+			</RevisionsListEmptyWrapper>
+		);
+	}
+
+	return (
+		<Container>
```

Comment:
The ui is wrong
in this branch:
<img width="1067" alt="image" src="https://github.com/3drepo/3drepo.io/assets/42032433/5c5b6344-c47f-4d92-aadf-504b225bea90">

In the mockup:

<img width="742" alt="image" src="https://github.com/3drepo/3drepo.io/assets/42032433/b43413db-013e-4319-8bd6-c223a65325f3">